http/src/main/java/org/http4s/blaze/http/parser/BodyAndHeaderParser.java

```diff
@@ -386,6 +387,7 @@ else if (ch == HttpTokens.LF) {
                     final int buff_size = in.remaining();
 
                     if (remaining_chunk_size <= buff_size) {
+                        // End of this chunk
```

Comment:
Pardon my foolishness here, and tangential nature. But doesn't it seem like this method is in dire need of more documentation.

This is literally how we handle all chunked content which is now rather common, and it seems our documentation is a few in line developer notes. `chunkedContent`.

Also how much is the performance benefit of keeping this in Java over Translating to equivalent Scala?